meta-facebook/yv35-cl/src/platform/plat_dimm.c

```diff
@@ -0,0 +1,447 @@
+		LOG_ERR("Failed to read I3C MUX status, ret=%d", ret);
+		return false;
+	}
+
+	if (GETBIT(i2c_msg.data[0], 0) == I3C_MUX_TO_BIC) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+bool is_dimm_present(uint8_t dimm_id)
+{
+	return dimm_data[dimm_id].is_present;
+}
+
+bool is_dimm_init()
+{
+	return is_dimm_data_init;
+}
+
+int all_brocast_ccc(I3C_MSG *i3c_msg)
+{
+	CHECK_NULL_ARG_WITH_RETURN(i3c_msg, -1);
+
+	int ret = 0;
+
+	ret = i3c_brocast_ccc(i3c_msg, I3C_CCC_RSTDAA, I3C_BROADCAST_ADDR);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = i3c_brocast_ccc(i3c_msg, I3C_CCC_SETAASA, I3C_BROADCAST_ADDR);
+	if (ret != 0) {
+		return ret;
+	}
+
+	return ret;
+}
+
+int get_pmic_error_raw_data(int dimm_index, uint8_t *data)
+{
+	CHECK_NULL_ARG_WITH_RETURN(data, -1);
+
+	int i = 0, fail_count = 0;
+
+	for (i = 0; i < sizeof(dimm_data[dimm_index].pmic_error_data); i++) {
+		if (dimm_data[dimm_index].pmic_error_data[i] == SENSOR_FAIL) {
+			fail_count++;
+		}
+	}
+
+	// PMIC error data read failed
+	if (fail_count == sizeof(dimm_data[dimm_index].pmic_error_data)) {
+		return -1;
+	}
+
+	memcpy(data, &dimm_data[dimm_index].pmic_error_data,
+	       sizeof(dimm_data[dimm_index].pmic_error_data));
+
+	return 0;
+}
+
+void get_pmic_power_raw_data(int dimm_index, uint8_t *data)
+{
+	CHECK_NULL_ARG(data);
+
+	memcpy(data, &dimm_data[dimm_index].pmic_pwr_data,
+	       sizeof(dimm_data[dimm_index].pmic_pwr_data));
+}
+
+void get_spd_temp_raw_data(int dimm_index, uint8_t *data)
+{
+	CHECK_NULL_ARG(data);
+
+	memcpy(data, &dimm_data[dimm_index].spd_temp_data,
+	       sizeof(dimm_data[dimm_index].spd_temp_data));
+}
+
+int sensor_num_map_dimm_id(uint8_t sensor_num)
+{
+	int dimm_id = -1;
+
+	switch (sensor_num) {
+	case SENSOR_NUM_PWR_DIMMA0_PMIC:
+	case SENSOR_NUM_TEMP_DIMM_A0:
+		dimm_id = DIMM_ID_A0;
+		break;
+	case SENSOR_NUM_PWR_DIMMA2_PMIC:
+	case SENSOR_NUM_TEMP_DIMM_A2:
+		dimm_id = DIMM_ID_A2;
+		break;
+	case SENSOR_NUM_PWR_DIMMA3_PMIC:
+	case SENSOR_NUM_TEMP_DIMM_A3:
+		dimm_id = DIMM_ID_A3;
+		break;
+	case SENSOR_NUM_PWR_DIMMA4_PMIC:
+	case SENSOR_NUM_TEMP_DIMM_A4:
+		dimm_id = DIMM_ID_A4;
+		break;
+	case SENSOR_NUM_PWR_DIMMA6_PMIC:
+	case SENSOR_NUM_TEMP_DIMM_A6:
+		dimm_id = DIMM_ID_A6;
+		break;
+	case SENSOR_NUM_PWR_DIMMA7_PMIC:
+	case SENSOR_NUM_TEMP_DIMM_A7:
+		dimm_id = DIMM_ID_A7;
+		break;
+	default:
+		dimm_id = -1;
+		break;
+	}
+
+	return dimm_id;
+}
+
+int pal_get_pmic_pwr(uint8_t sensor_num, uint8_t *data)
+{
+	CHECK_NULL_ARG_WITH_RETURN(data, -1);
+
+	int dimm_id = -1;
+
+	dimm_id = sensor_num_map_dimm_id(sensor_num);
+	if (dimm_id < 0) {
+		return -1;
+	}
+
+	get_pmic_power_raw_data(dimm_id, (uint8_t *)data);
+
+	// If sensor data is SENSOR_FAIL, return failed
+	if (data[0] == SENSOR_FAIL) {
+		return -1;
+	}
+
+	return 0;
+}
+
+int pal_get_spd_temp(uint8_t sensor_num, uint8_t *data)
+{
+	CHECK_NULL_ARG_WITH_RETURN(data, -1);
+
+	int dimm_id = -1;
+
+	dimm_id = sensor_num_map_dimm_id(sensor_num);
+	if (dimm_id < 0) {
```

Comment:
dimm_id is defined as uint8_t so no need to check for negative value.

Reply:
We add new definition "DIMM_ID_UNKNOWN" to check if dimm_id can't map.